src/server/frontend_xwayland/xwayland_wm.h

```diff
@@ -113,25 +89,18 @@ class XWaylandWM
 
     std::mutex mutex;
 
-    // Cursor
-    xcb_cursor_t xcb_cursor_image_load_cursor(const XcursorImage *img);
-    xcb_cursor_t xcb_cursor_images_load_cursor(const XcursorImages *images);
-    xcb_cursor_t xcb_cursor_library_load_cursor(const char *file);
-
     std::shared_ptr<WaylandConnector> const wayland_connector;
     std::shared_ptr<dispatch::MultiplexingDispatchable> const dispatcher;
     wl_client* const wayland_client;
     std::shared_ptr<XWaylandWMShell> const wm_shell;
+    std::unique_ptr<XWaylandCursors> cursors; ///< Only non-const so it can be reset in the destructor
```

Comment:
(Non blocking) "non-const so it can be reset in the destructor" sounds as though we have our life-time management wrong. (Also, on a cursory examination of current `master` I think this is now unnecessary.)

Reply:
Correct on both counts. This was working around a life-time management issue, which has now been fixed on master. I rebased this branch on that and made cursors const.